tools/CMakeLists.txt

```diff
@@ -1,5 +1,3 @@
-CHECK_COMPILER_FLAG_AND_ENABLE_IT(-fno-strict-aliasing)
-
```

Comment:
From the commit message, it seems you are confusing `-Wno-strict-aliasing` and `-fno-strict-aliasing`. The first silents warnings about strict aliasing, but the second allows the compiler to use strict aliasing rules for optimizations (https://gcc.gnu.org/onlinedocs/gcc-4.9.2/gcc/Optimize-Options.html).

Removing this option does not just activate warnings, it also enables potentially dangerous optimizations. Eliminating the warnings does not guarantee that other optimizations are safe. To me, the potential danger of activating strict aliasing is too high compared to the potential performance benefit (which is probably negligible actually, but without numbers this is pure speculation on my part).

Reply:
As the strict aliasing issues have been fixed, we can turn that on. I've fixed the commit message.

Reply:
The *warnings* have been fixed, but how can you be sure that all the *issues* have been fixed? Warnings are not meant to be exhaustive (the doc actually says explicitly "The warning does not catch all cases", https://gcc.gnu.org/onlinedocs/gcc-4.9.2/gcc/Warning-Options.html#Warning-Options). If the compiler could warn about all dangerous cases, it would mean that it would be able to do exact alias analysis, and there would be no point using the strict aliasing rule.

Reply:
But how should we find and fix the source if we hide/disable it?